tests/integration/sugar/test_wallet.py

```diff
@@ -37,52 +38,129 @@ async def test_generate_faucet_wallet_custom_host_async_websockets(self):
         with WebsocketClient("wss://xls20-sandbox.rippletest.net:51233") as client:
-            wallet = sync_generate_faucet_wallet(
-                client, faucet_host="faucet-nft.ripple.com"
+            sync_generate_faucet_wallet_and_fund_again(
+                self, client, "faucet-nft.ripple.com"
             )
-            result = client.request(
+
+    def test_generate_faucet_wallet_custom_host_sync_json_rpc(self):
+        client = JsonRpcClient("http://xls20-sandbox.rippletest.net:51234")
+        sync_generate_faucet_wallet_and_fund_again(
+            self, client, "faucet-nft.ripple.com"
+        )
+
+    async def test_generate_faucet_wallet_testnet_async_websockets(self):
+        async with AsyncWebsocketClient(
+            "wss://s.altnet.rippletest.net:51233"
+        ) as client:
+            await generate_faucet_wallet_and_fund_again(self, client)
+
+    async def test_generate_faucet_wallet_devnet_async_websockets(self):
+        async with AsyncWebsocketClient(
+            "wss://s.devnet.rippletest.net:51233"
+        ) as client:
+            await generate_faucet_wallet_and_fund_again(self, client)
+
+    async def test_generate_faucet_wallet_nft_devnet_async_websockets(self):
+        async with AsyncWebsocketClient(
+            "ws://xls20-sandbox.rippletest.net:51233"
+        ) as client:
+            await generate_faucet_wallet_and_fund_again(self, client)
+
+    async def test_generate_faucet_wallet_amm_devnet_async_websockets(self):
+        async with AsyncWebsocketClient(
+            "wss://amm.devnet.rippletest.net:51233"
+        ) as client:
+            await generate_faucet_wallet_and_fund_again(self, client)
+
+    async def test_generate_faucet_wallet_hooks_v2_testnet_async_websockets(self):
+        async with AsyncWebsocketClient(
+            "wss://hooks-testnet-v2.xrpl-labs.com"
+        ) as client:
+            wallet = await generate_faucet_wallet(client)
+            result = await client.request(
                 AccountInfo(
                     account=wallet.classic_address,
                 ),
             )
-            self.assertTrue(int(result.result["account_data"]["Balance"]) > 0)
+            balance = int(result.result["account_data"]["Balance"])
+            self.assertTrue(balance > 0)
 
-    def test_generate_faucet_wallet_custom_host_sync_json_rpc(self):
-        client = JsonRpcClient("http://xls20-sandbox.rippletest.net:51234")
-        wallet = sync_generate_faucet_wallet(
-            client, faucet_host="faucet-nft.ripple.com"
-        )
-        result = client.request(
-            AccountInfo(
-                account=wallet.classic_address,
-            ),
-        )
-        self.assertTrue(int(result.result["account_data"]["Balance"]) > 0)
+    # Named different from test_generate_faucet_wallet_hooks_v2_testnet_async_websockets
+    # so the test runs far from each other since hooks v2 testnet faucet
+    # requires 10 seconds between calls
```

Comment:
This seems like it'd be very flaky. Can you add in an explicit variable to ensure that we're waiting at least long enough? 

Ex. A variable like `last_hooks_faucet_call` and then use it as a guard to sleep if you accidentally start the test too early? (E.g. "We interacted with the hooks faucet 4 seconds ago, let's sleep for 6 seconds")

Reply:
(This variable should be used to protect both tests in case they eventually are run in a different order)